sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,299 @@
+                    }
+                }
+            };
+
+            performWorkload(baseSubscriber, i);
+        }
+        synchronized (count) {
+            while (count.get() < i) {
+                count.wait();
+            }
+        }
+
+        long endTime = System.currentTimeMillis();
+        logger.info("[{}] operations performed in [{}] seconds.",
+            configuration.getNumberOfOperations(), (int) ((endTime - startTime) / 1000));
+
+        reporter.report();
+        reporter.close();
+    }
+
+    void shutdown() {
+        for (CosmosAsyncClient asyncClient : clientDocsMap.keySet()) {
+            asyncClient.close();
+        }
+    }
+
+    protected void onSuccess() {
+    }
+
+    protected void onError(Throwable throwable) {
+    }
+
+    private PojoizedJson generateDocument(String idString, String dataFieldValue, String partitionKey) {
+        com.azure.cosmos.benchmark.PojoizedJson instance = new com.azure.cosmos.benchmark.PojoizedJson();
+        Map<String, String> properties = instance.getInstance();
+        properties.put("id", idString);
+        properties.put(partitionKey, idString);
+
+        for (int i = 0; i < configuration.getDocumentDataFieldCount(); i++) {
+            properties.put("dataField" + i, dataFieldValue);
+        }
+
+        return instance;
+    }
+
+    private boolean shouldContinue(long startTimeMillis, long iterationCount) {
+
+        Duration maxDurationTime = configuration.getMaxRunningTimeDuration();
+        int maxNumberOfOperations = configuration.getNumberOfOperations();
+
+        if (maxDurationTime == null) {
+            return iterationCount < maxNumberOfOperations;
+        }
+
+        if (startTimeMillis + maxDurationTime.toMillis() < System.currentTimeMillis()) {
+            return false;
+        }
+
+        if (maxNumberOfOperations < 0) {
+            return true;
+        }
+
+        return iterationCount < maxNumberOfOperations;
+    }
+
+    private void createClients() {
+        String csvFile = "clientHostAndKey.txt";
+        BufferedReader br = null;
+        String line = "";
+        String splitBy = ",";
+        try {
+            br = new BufferedReader(new FileReader(csvFile));
+            while ((line = br.readLine()) != null) {
+                String[] hostAndKey = line.split(splitBy);
+                if (hostAndKey.length == 2) {
+                    CosmosAsyncClient asyncClient = new CosmosClientBuilder()
+                        .endpoint(hostAndKey[0])
+                        .key(hostAndKey[1])
+                        .connectionPolicy(configuration.getConnectionPolicy())
+                        .consistencyLevel(configuration.getConsistencyLevel())
+                        .connectionReuseAcrossClientsEnabled(true)
+                        .buildAsyncClient();
+                    List<PojoizedJson> docsToRead = new ArrayList<>();
+                    if (!configuration.isDeleteCollections()) {
```

Comment:
it seems you are using `isDeleteCollections` for not only deleting collections but also for creating collection?

is that right? the config seems to be overloaded.

Reply:
during deleteCollections workload we don't want to create db/collection/data . createClient method during initialization will create client/db/container/data.

Reply:
discussed offline